src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -592,15 +604,19 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
           capture_event.mutable_gpu_queue_submission();
 
       WriteMetaInfo(completed_submission.meta_information, submission_proto->mutable_meta_info());
-      WriteCommandBufferTimings(completed_submission, submission_proto);
-      WriteDebugMarkers(completed_submission, submission_proto);
+      bool has_command_buffer_timestamps =
+          WriteCommandBufferTimings(completed_submission, submission_proto);
+      bool has_debug_marker_timestamps = WriteDebugMarkers(completed_submission, submission_proto);
 
-      if (vulkan_layer_producer_ != nullptr) {
+      if (vulkan_layer_producer_ != nullptr &&
```

Comment:
Not related to this change directly, but what is actually the case when vulkan_layer_producer_ is nullptr? In that case we never send any events, so all work done in the layer doesn't actually have any outcome. Can we early out in this function (or even earlier?) when vulkan_layer_producer_ is nullptr or do we still need to do the work to maintain the state? In other words, can this be a CHECK here with graceful error handling done earlier?

Reply:
@dpallotti correct me if I am wrong, but I think this may only happen if the producer actually dies. Otherwise, we should not end up here. Might be, that we can even turn this into a CHECK.

In that case, I guess we need to do the work, especially to reset the timer slots.

Reply:
> I think this may only happen if the producer actually dies

What do you mean with the producer dying? I guess that if the producer crashes the whole game crashes, as it's the same process.

Either way, I'm not really sure about the conditions on `vulkan_layer_producer_` in `SubmissionTracker`. I guess we kept it generic and support both cases of set and not set. In particular because `vulkan_layer_producer_` is set on `OnCreateInstance` and removed on `OnDestroyInstance`.

Reply:
This should be however, in a different PR.